pkg/apiserver/registry/system/supportbundle/rest.go

```diff
@@ -133,17 +133,18 @@ func (r *supportBundleREST) Create(ctx context.Context, obj runtime.Object, _ re
 	ctx, cancelFunc := context.WithCancel(context.Background())
 	r.cache = &systemv1beta1.SupportBundle{
 		ObjectMeta: metav1.ObjectMeta{Name: r.mode},
+		Days:       requestBundle.Days,
 		Status:     systemv1beta1.SupportBundleStatusCollecting,
 	}
 	r.cancelFunc = cancelFunc
-
+	cacheCopy := r.cache.DeepCopy()
```

Comment:
why it needs a copy?

Reply:
We need to tell the days to the collect function. Although for now, this REST does not use an internal struct, in the future we may add more options to it and thus copy the struct instead of separate options arguments can reduce code changes.

Reply:
I meant why don't pass `r.cache` to `r.collectAgent` directly but make a copy first?
If it's for race condition consideration, `r.cache` can be override by another call between L138-L140 as well?
Should it be:
```
request := &systemv1beta1.SupportBundle{...}
r.cache = request
...
b, err = r.collectAgent(ctx, request)
```

Reply:
Thanks, @tnqn. You're right the copy is used to avoid a race condition. But since we acquire the lock at `:127`, I don't think there will be any overwriting issue?

Reply:
Is the race condition because of L154? Better to comment what the copy is preventing from as there are a few goroutines here making it a little difficult to understand the potential race condition.
And could it just pass `requestBundle.Days` to `collectAgent` to avoid copy at all.

Reply:
Make sense, I have removed the deep copy.